### DISCUSSION TITLE
Piggy Bank overfill

### DISCUSSION THREAD

**Discussion by AlexKalopsia** [firefly-iii/firefly-iii#6411] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

I would like to have the option to overfill a piggy bank.

I use piggy banks to keep track of year savings. 
Say that I set year savings for 20000 euros. 
Let's say that in October I reached the goal but I want to keep saving in the piggy bank. 

Right now, if I select the piggy bank, the money won't be added (I won't even receive a warning of any sort).
I either have to manually expand the piggy bank before registering the transaction, or I have to make a new piggy bank for the etxra savings.

I currently don't really see why this is not allowed, but would be happy to know more and understand why this is the case. 

### Solution

Piggy bank overfill would allow me to see "how much more than I planned" I managed to save. 
For instance, in January I set 20000 euros goal, but at the end of the year I can see I managed to get it to 120% (20% more than I planned). I think this is just more flexible and doesn't seem to interfere with other systems.

**Comment by JC5** [comment]
It's a piggy bank. The goal is to fill it. If you want to add more, expand the piggy bank or set it to zero (no goal).

**Reply by timaschew** [reply]
> expand the piggy bank or set it to zero (no goal).

Set it to zero sounds ok to me. The description could reflect the goal as well.

**Comment by AlexKalopsia** [comment]
As I said, I think being able to have info on "how much more than planned" you managed to put in the piggy bank is valuable info.

In terms of UX, it took me a while to understand what was even happening. The piggy bank was not showing my savings anymore, even though I kept adding money to it. There is definitely some lack of clarity. The way I see it, there are a few options:

- Allow overfill: Money goes past the max, and shows >100% status
- Auto-expand: Extend the upper cap if I keep adding money
- Block: Just don't allow to connect to piggy bank if it reaches and overfills the cap

As it is right now, information will just be lost?

**Comment by JC5** [comment]
> As I said, I think being able to have info on "how much more than planned" you managed to put in the piggy bank is valuable info.

I disagree. 

I don't mind hiding full piggy banks, or marking them as such. That's a good idea. I'm not going to add these options you suggest.

**Comment by AlexKalopsia** [comment]
I don't fully understand the thinking here, but let's assume we don't want overfill, ok! 

If we *hide* a piggy bank from the New Transaction page, from the user pov this will surely be perceived as a bug?

With "marking them as such" do you mean just having some form of "FULL" label? Because - unless the transaction is blocked (option 3), you would still end up with the user thinking they added money to a piggy bank, but that actually didn't correctly work? 

ie

Piggy Bank: 19000/20000
Add 2000 to Piggy Bank

1000 would be lost in the vacuum?

Or are you suggesting that I would have to make a split, and manually make sure I only add 1000 to it?

**Comment by JC5** [comment]
> 1000 would be lost in the vacuum?

Why do you consider this money to be lost?

**Comment by AlexKalopsia** [comment]
To me putting the money in my piggy bank translates to

"Put my money into my savings goal for the year"

If I add 2000 to my savings piggy bank, it will actually only add 1000 (as it reaches the cap), right?

**Reply by JC5** [reply]
Yes, and the money is still in your account. You can see how much you have left for piggy banks on the bottom of the piggy banks page, so you can spend it elsewhere.
